### DISCUSSION TITLE
Section 3: Anyone know the correct way to run solidity-code-metrics in intellij in windows system

### DISCUSSION THREAD

**Discussion by Vincent71399** [Cyfrin/security-and-auditing-full-course-s23#205] [Q&A]
what I did is I installed solidity-code-metrics via npm/yarn in the project of 3-passwordstore audit
from this repo https://github.com/Consensys/solidity-metrics

then I use the command: solidity-code-metrics src/PasswordStore.sol > report/PasswordStore.md
the .md file can be generated, but all the charts are not displayed

the following is the .md file I got
[PasswordStore.md](https://github.com/user-attachments/files/16093951/PasswordStore.md)
I checked the md code, they are like 
| File   |
| ------ |
| None |

| File   | Contract | Doppelganger | 
| ------ | -------- | ------------ |

and those plus signs are not clickable

**Comment by EngrPips** [comment]
I use `solidity-metrics` in `vscode`. by downloading the extension. When I right-click on the file I want to use it for, it brings a list of options, and among those options is `solidity metrics.` I just click on it, and it generates a PDF that contains all the information about the file

<img width="1440" alt="Screenshot 2024-07-04 at 07 57 04" src="https://github.com/Cyfrin/security-and-auditing-full-course-s23/assets/95760918/fd16bf31-1221-4e99-a503-0ac4a37d53cf">

The PDF it generates looks like the one below.

<img width="1440" alt="Screenshot 2024-07-04 at 07 59 45" src="https://github.com/Cyfrin/security-and-auditing-full-course-s23/assets/95760918/7d1be70a-c05e-44a8-a42b-c1467dae253d">

**Reply by Vincent71399** [reply]
Thx for sharing, but I intended to stick to Intellij

**Reply by EngrPips** [reply]
Oh, I see.

**Reply by giriraj-kulkarni** [reply]
hey whenever i do that using vs code or vs codium it shows no valid solidity source found error

**Reply by EngrPips** [reply]
I am unsure what could be the issue.
